pyam/core.py

```diff
@@ -343,24 +343,31 @@ def pivot_table(self, index, columns, values='value',
                             aggfunc=aggfunc, fill_value=fill_value)
         return df
 
-    def interpolate(self, year):
+    def interpolate(self, time):
         """Interpolate missing values in timeseries (linear interpolation)
 
         Parameters
         ----------
-        year: int
-             year to be interpolated
+        time: int, datetime
+             Time or year to be interpolated. This must match the
+             date-time/year style of self.
         """
-        df = self.pivot_table(index=IAMC_IDX, columns=['year'],
+        df = self.pivot_table(index=IAMC_IDX, columns=[self.time_col],
                               values='value', aggfunc=np.sum)
-        # drop year-rows where values are already defined
-        if year in df.columns:
-            df = df[np.isnan(df[year])]
-        fill_values = df.apply(fill_series,
-                               raw=False, axis=1, year=year)
+        # drop time-rows where values are already defined
+        if time in df.columns:
+            df = df[np.isnan(df[time])]
+        if self.time_col == "year":
```

Comment:
it would be more elegant to move this sanity-check as first operation of this function (before the pivot), then it is not necessary to do the if-then statement at all (I think)

```
if self.time_col == "year" and not isinstance(time, int):
    msg = 'the `time` argument `{}` is not an integer'.format(time)
    raise ValueError(msg.format(time))
```
Added side bonus - the error is raised faster (not first do the pivot, then check).

PS: also subtle change above for consistency - trying to avoid `"` in favour of `'`